### PR TITLE
Update loom and gradle

### PR DESCRIPTION
**Pull request by Wyvest**
Updates loom to 1.2, gradle to 8.1.1, and OneConfig to 0.2.0.

Essential's fork currently needs a lot of fixes that are waiting to be merged, so we use the Polyfrost fork

### CODE REVIEW

File: oneconfigquarantine/build.gradle.kts [hunk 1]
```diff
@@ -42,7 +42,7 @@ dependencies {
 		mappings("de.oceanlabs.mcp:mcp_stable:22-1.8.9")
 		forge("net.minecraftforge:forge:1.8.9-11.15.1.2318-1.8.9")
 
-		modApi("cc.polyfrost:oneconfig-1.8.9-forge:0.1.0-alpha+") // Don't you just love 0.1.0-alpha+
+		modApi("cc.polyfrost:oneconfig-1.8.9-forge:0.2.0-alpha+") // no
```
Comment (romangraef) [inline oneconfigquarantine/build.gradle.kts:45]: Yeah please stop making that the only way of including your mod and allow us to use proper versions. You are lowkey blocking us from using dependency verification to verify we arent pulling in maliciously modified updates. Just have a `1.0.0` version and increment that like a normal person. This is disgraceful for a project that *aims to be a library*.

**Review by romangraef — CHANGES_REQUESTED** [review]
Honestly not really a big fan of this PR. It's entirely pointless (does not give a performance benefit (in fact a `./gradlew clean build` seems to be slower on my machine, even after downloading all dependencies: 2 minutes (on the second ./gradlew clean build) vs 59 seconds with old essential loom)) and you don't use any of the features in loom 1.2 to warrant this update. I'd rather just use unimined 1.0.0 once that is released, since it is both faster, more flexible and not affiliated with essential (who is a known evil). Also: even tho i think loom 1.2 is faster to compile than loom 0.10, essentials loom is slower somehow, and i dont want to put effort debugging that for a project i dont support in the first place.

File: oneconfigquarantine/build.gradle.kts (hunk 1 above)
Comment (Ascynx) [reply oneconfigquarantine/build.gradle.kts:45]: too much trolling

File: src/main/java/io/github/moulberry/notenoughupdates/mixins/MixinThreadDownloadImageDataThread.java [hunk 2]
```diff
@@ -37,7 +37,8 @@ public class MixinThreadDownloadImageDataThread {
 			value = "INVOKE",
 			target = "Ljava/net/HttpURLConnection;setDoOutput(Z)V"
 		),
-		locals = LocalCapture.CAPTURE_FAILSOFT
+		locals = LocalCapture.CAPTURE_FAILSOFT,
+		remap = false
```
Comment (Ascynx) [inline src/main/java/io/github/moulberry/notenoughupdates/mixins/MixinThreadDownloadImageDataThread.java:41]: It is under ``net.minecraft`` tho, pretty sure most if not all class files under that package are actually mapped.
If it was intended for the invoked target it should've been added in the ``@At`` interface and not outside.
But it does not seem to actually change anything.

**Comment by Wyvest** [comment]
> Honestly not really a big fan of this PR. It's entirely pointless (does not give a performance benefit (in fact a `./gradlew clean build` seems to be slower on my machine, even after downloading all dependencies: 2 minutes (on the second ./gradlew clean build) vs 59 seconds with old essential loom)) and you don't use any of the features in loom 1.2 to warrant this update. I'd rather just use unimined 1.0.0 once that is released, since it is both faster, more flexible and not affiliated with essential (who is a known evil). Also: even tho i think loom 1.2 is faster to compile than loom 0.10, essentials loom is slower somehow, and i dont want to put effort debugging that for a project i dont support in the first place.

Fair enough, will see what other people think though. Personally I just wanted to update it for the sake of being up to date which now I think about it isn't the best option

File: src/main/java/io/github/moulberry/notenoughupdates/mixins/MixinThreadDownloadImageDataThread.java (hunk 2 above)
Comment (Wyvest) [reply src/main/java/io/github/moulberry/notenoughupdates/mixins/MixinThreadDownloadImageDataThread.java:41]: This is a java function that is being overridden, and removing this breaks compiling.

**Pull request closed without merging**